### PR TITLE
Jvenstad/pushing build system

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please review.

Must be merged with https://git.ouroath.com/vespa/hosted/pull/5181

Apologies for the messy commit story >_< I was a bit over-ambitious last year ...

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/BuildSystem.java [hunk 1]
```diff
@@ -1,34 +0,0 @@
...
-import com.yahoo.vespa.hosted.controller.api.integration.BuildService.BuildJob;
-import com.yahoo.vespa.hosted.controller.application.DeploymentJobs.JobType;
-
-import java.util.List;
-
-/**
- * @author jvenstad
- * @author mpolden
- */
-public interface BuildSystem {
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/BuildSystem.java]: No need to interface a queue, so the interface is gone.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentQueue.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import java.util.Optional;
+import java.util.function.Consumer;
+
+/**
+ * Stores a queue for each type of job, and offers jobs from each of these to a periodic
+ * polling mechanism which is responsible for triggering the offered jobs in an external build service.
+ *
+ * @author jvenstad
+ */
+public class DeploymentQueue {
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentQueue.java]: I guess git didn't catch the rename -- this was called PolledBuildSystem before.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerer.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.logging.Logger;
+
+/**
+ * Triggers deployment jobs in an external BuildService.
+ *
+ * Triggering is performed by an Executor, as there is no guarantee the BuildService provides a timely response.
+ *
+ * @author jvenstad
+ */
+public class DeploymentTriggerer extends Maintainer {
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerer.java]: This replaced the hosted thing.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/screwdriver/ScrewdriverApiHandler.java [hunk 4]
```diff
@@ -95,7 +95,7 @@ private HttpResponse post(HttpRequest request) {
     private HttpResponse delete(HttpRequest request) {
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/screwdriver/ScrewdriverApiHandler.java]: Ah, this should probably have been removed.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTester.java [hunk 5]
```diff
@@ -69,9 +69,7 @@ public DeploymentTester(ControllerTester tester) {
 
     public ApplicationController applications() { return tester.controller().applications(); }
 
-    // TODO: This thing simulates the wrong thing: the build system won't hold the jobs that are running,
-    // and so these should be consumed immediately upon triggering, and be "somewhere else" while running.
```
Comment (jonmv) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTester.java]: I wanted to do this, and with a MockTimeline, but that'll be later (or never)...

**Review by bratseth — APPROVED** [review]
👍

**Pull request merged**